source/support/rotate-secrets/index.html.md.erb

```diff
@@ -0,0 +1,180 @@
+[GOV.UK Notify API key][].
+
+Secrets are referenced in the Concourse configuration using `((secret
+name))`. For example with the daily statistics task, the
+`report-bearer-token` is a secret made available to the task through
+the `BEARER_TOKEN` environment variable:
+
+```yaml
+- task: trigger-bigquery-export
+  file: git-main/concourse/tasks/trigger-bigquery-export.yml
+  params:
+    CDN_DOMAIN: account.publishing.service.gov.uk
+    BEARER_TOKEN: ((report-bearer-token))
+```
+
+To list all secrets for the GOV.UK Concourse team, run the following in the gds-cli:
+
+```
+$ gds cd secrets ls cd-govuk-tools
+```
+
+[the daily statistics task]: https://github.com/alphagov/govuk-account-manager-prototype/blob/7e69b8cb8daec1e6e23d32374df3f8a1fa45b477/concourse/pipeline.yml#L249-L271
+[GOV.UK Notify API key]: https://github.com/alphagov/govuk-account-manager-prototype/blob/5443585c3d2dd5d399efc18f126f0bdf831036b5/concourse/tasks/deploy-to-govuk-paas.yml#L59
+
+### Rotating a secret
+
+1. Generate a new secret value, which varies per secret.
+
+2. Set the new value using the gds-cli:
+
+    ```
+    $ gds cd secrets add cd-govuk-tools <pipeline-name>/<secret-name> "<value>"
+    ```
+
+3. If this is a secret used by an app, deploy the app.
+
+## How to rotate the secrets we have
+
+In general you rotate secrets by generating a new value and
+setting the value with the gds-cli. You must use a different method for the following secrets.
+
+### Secrets used by multiple pipelines
+
+#### `grafana-api-key`
+
+Generate a new API key in [Grafana][], and deactivate the old API key.
+
+[Grafana]: https://grafana-paas.cloudapps.digital/org/apikeys
+
+#### `paas-username` and `paas-password`
+
+Change the password by [triggering a password reset][], which gets sent to the govuk-accounts-developers Google Group.
+
+[triggering a password reset]: https://admin.london.cloud.service.gov.uk/password/request-reset
+
+#### `sentry-dsn`
+
+Generate a new DSN from the Sentry project settings page.
+
+#### `slack_webhook_url`
+
+Tell IT Support that our Slack webhook has been leaked, and that we
+need a new one.
+
+### Secrets used by the account manager
+
+#### `basic-auth-username` and `basic-auth-password`
+
+These are arbitrary strings. After changing them, update the
+"deployed applications" Trello card.
+
+#### `bigquery-credentials-production`
+
+Generate new JSON credentials in the [Google Cloud Console][], and
+deactivate the old credentials.
+
+[Google Cloud Console]: https://console.cloud.google.com/iam-admin/serviceaccounts?orgonly=true&project=govuk-account-analysis-db&supportedpurview=project
+
+#### `notify-api-key-production` and `notify-api-key-staging`
+
+Generate a new API key from [GOV.UK Notify][], and deactivate the old
+API key.
+
+[GOV.UK Notify]: https://www.notifications.service.gov.uk/
+
+#### `oidc-pepper-production` and `oidc-pepper-staging`
+
+Generate an arbitrary string using `rake secret`.
```

Comment:
run this in gds-cli? or in some other way?

Reply:
`rake` is a tool to run maintenance tasks for the apps, and one of the built-in tasks is this `secret` one which generates a random long string.  So to run this a developer would need a copy of the app code on their machine.